hls-plugin-api/src/Ide/Plugin/Config.hs

```diff
@@ -54,18 +61,22 @@ data Config =
     , plugins            :: !(Map.Map T.Text PluginConfig)
     } deriving (Show,Eq)
 
-instance Default Config where
-  def = Config
+-- | Default configuration values
```

Comment:
Maybe expand this comment a bit to explain the general approach? Given that we were both complaining about finding this bit hard to understand, maybe let's try to make the new stuff easy to follow :) perhaps something like:

```
Default configuration values. The default values for plugins can be passed by the caller so that plugin definitions can provide their own, custom, default values for configuration values.